src/session-pool.ts

```diff
@@ -681,25 +766,49 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
   /**
    * Borrows the first session from the inventory.
    *
+   * @param {boolean} longRunningTransaction Is transaction long-running.
    * @private
    *
    * @return {Session}
    */
-  _borrowFrom(): Session {
+  _borrowFrom(longRunningTransaction: boolean): Session {
     const session = this._inventory.sessions.pop()!;
     this._inventory.borrowed.add(session);
+    session.lastUsed = Date.now();
+    session.longRunningTransaction = longRunningTransaction;
+
+    const maxSessions = this.maxSessions;
+
+    // Check if more than 95% of the pool is exhausted and if cleanup is required.
+    if (
+      (this.options.closeInactiveTransactions || this.options.logging) &&
+      this.borrowed / maxSessions >= 0.95
+    ) {
+      this._startCleaningLongRunningSessions();
+    }
+
+    // Log if entire session pool is exhausted.
+    if (this.options.logging && this.borrowed / maxSessions === 1.0) {
+      this.database.logger?.log(
+        'info',
+        '100% of the session pool is exhausted.' +
+          'Increase max sessions in Session Pool Options as per requirement.'
```

Comment:
```suggestion
          ` Increase max sessions in SessionPoolOptions if your application needs to execute more than ${maxSessions} transactions in parallel. Check your application for session leaks if this warning is unexpected, for example if your application would normally not execute more than ${maxSessions} transactions in parallel.`
```

And maybe also add hints on 'how to check for session leaks' in the above warning? Maybe even log the traces of the currently checked out sessions?